### PR TITLE
Fix the `failed to load test execution class` error

### LINKED ISSUE #38614: [Bug]: Getting `failed to load test execution class` error when running test with --native flag
### Description

Getting the following error time to time, when running the graphql module test with `--native` flag 

```
Running Tests

	graphql.parser
error: failed to load test execution class :ballerina.graphql$0046parser$test.1.tests.test_execute-generated_1
error: there are test failures
```

Github action logs related to this error can be found here: https://github.com/ballerina-platform/module-ballerina-graphql/actions/runs/3430907608/jobs/5718422153#step:9:194 

### Steps to Reproduce

_No response_

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Test Framework

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by Thevakumar-Luheerathan**
Fixes #38614 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunNativeImageTestTask.java [hunk 1]
```diff
@@ -180,6 +180,14 @@ public void execute(Project project) {
         }
 
         TestUtils.writeToTestSuiteJson(testSuiteMap, testsCachePath);
+        Path nativeConfigPath = null;
+        try {
+            nativeConfigPath = target.getNativeConfigPath();
```
Comment (Ibaqu) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunNativeImageTestTask.java:185]: Any reason why we are resolving the nativeConfigPath here? Its already resolved in Ln307 previously. Seems like the `createReflectionConfig` has just been moved up a few lines. 

Does it have an impact?

Comment (Thevakumar-Luheerathan) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunNativeImageTestTask.java:185]: There is no specific reason for moving. I just wanted to use the `testSuiteMap` with that method.

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunNativeImageTestTask.java [hunk 2]
```diff
@@ -296,8 +304,7 @@ private int runTestSuiteWithNativeImage(Package currentPackage, JBallerinaBacken
         Path nativeConfigPath = target.getNativeConfigPath();   // <abs>target/cache/test_cache/native-config
         Path nativeTargetPath = target.getNativePath();         // <abs>target/native
 
-        // Create Configs
-        createReflectConfig(nativeConfigPath, currentPackage);
+
```
Comment (Ibaqu) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunNativeImageTestTask.java:307]: Remove this extra line

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java [hunk 3]
```diff
@@ -57,6 +61,8 @@ public static void createReflectConfig(Path nativeConfigPath, Package currentPac
 
         for (Module module : currentPackage.modules()) {
             String name = module.moduleName().toString();
+            String moduleName = ProjectUtils.getJarFileName(module);
+
```
Comment (Ibaqu) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java:65]: Remove extra line

**Review by Ibaqu — DISMISSED** [review]

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunNativeImageTestTask.java [hunk 4]
```diff
@@ -180,6 +180,14 @@ public void execute(Project project) {
         }
 
         TestUtils.writeToTestSuiteJson(testSuiteMap, testsCachePath);
+        Path nativeConfigPath = null;
+        try {
+            nativeConfigPath = target.getNativeConfigPath();
+            createReflectConfig(nativeConfigPath, project.currentPackage(), testSuiteMap);
+        } catch (IOException e) {
+            throw createLauncherException("error while generating the necessary reflection config", e);
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunNativeImageTestTask.java:188]:
```suggestion
            throw createLauncherException("error while generating the necessary graalvm reflection config", e);
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunNativeImageTestTask.java [hunk 5]
```diff
@@ -180,6 +180,14 @@ public void execute(Project project) {
         }
 
         TestUtils.writeToTestSuiteJson(testSuiteMap, testsCachePath);
+        Path nativeConfigPath = null;
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunNativeImageTestTask.java:183]: We can remove this redundant initialization.

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java [hunk 6]
```diff
@@ -91,28 +96,29 @@ public static void createReflectConfig(Path nativeConfigPath, Package currentPac
...
-                                    "io.ballerina.runtime.api.values.BString",
-                                    "io.ballerina.runtime.api.values.BString",
-                                    "io.ballerina.runtime.api.values.BString",
-                                    "io.ballerina.runtime.api.values.BString",
-                                    "io.ballerina.runtime.api.values.BString",
-                                    "io.ballerina.runtime.api.values.BString"
-                            }
-                    )
-            );
+            ReflectConfigClass testTestExecuteGenerated = new ReflectConfigClass("");
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java:99]: The variable name can be improved here. The IDE would suggest the variable name.
```suggestion
            ReflectConfigClass reflectConfigClass = new ReflectConfigClass("");
```

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java [hunk 7]
```diff
@@ -91,28 +96,29 @@ public static void createReflectConfig(Path nativeConfigPath, Package currentPac
...
+            if (testSuiteMap.get(moduleName) != null) {
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java:100]: Have we added entries with null values to the map? We should be able to use `testSuiteMap.containsKey`

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java (hunk 6 above)
Comment (Thevakumar-Luheerathan) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java:99]: There are few more `ReflectConfigClass` instances are created in same `NativeUtils.java`. So, The naming is based on how it is utilised(Each instance represents a class in `ReflectionConfig`).

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java [hunk 8]
```diff
@@ -57,84 +61,86 @@ public static void createReflectConfig(Path nativeConfigPath, Package currentPac
 
         for (Module module : currentPackage.modules()) {
             String name = module.moduleName().toString();
+            String moduleName = ProjectUtils.getJarFileName(module);
 
-            ReflectConfigClass testInitClass = new ReflectConfigClass(getQualifiedClassName(org, name, version,
+            ReflectConfigClass testInitRefConfCls = new ReflectConfigClass(getQualifiedClassName(org, name, version,
```
Comment (azinneera) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java:66]:
```suggestion
            ReflectConfigClass testInitRefConfClz = new ReflectConfigClass(getQualifiedClassName(org, name, version,
```

Comment (azinneera) [reply cli/ballerina-cli/src/main/java/io/ballerina/cli/utils/NativeUtils.java:66]: Address other places as well

**Review by azinneera — DISMISSED** [review]

**Review by azinneera — APPROVED** [review]

**Pull request merged**